pallets/committee/src/lib.rs

```diff
@@ -325,6 +329,13 @@ pub mod pallet {
 				proposals.retain(|hash| if let Some(votes) = Self::get_votes_for(hash) { votes.end > n } else { false })
 			});
 
+			// reset the voting period if has pending voting period
+			if let Some(pending_voting_period) = PendingVotingPeriod::<T>::get() {
+				writes = writes.saturating_add(pending_voting_period.encode().len() as Weight);
```

Comment:
why do we `encode`here? this is a single additional write, the `VotingPeriod::<T>::set`

Reply:
oops, considered it determined by the data length